api/v1alpha1/bundle_types.go

```diff
@@ -18,48 +18,73 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	runtime "k8s.io/apimachinery/pkg/runtime"
 )
 
-// Source locates content to stage.
-type Source string
+type ProvisionerID string
+type BundleConditionType string
+
+const (
+	TypeUnpacked = "Unpacked"
+
+	ReasonUnpackPending    = "UnpackPending"
+	ReasonUnpacking        = "Unpacking"
+	ReasonUnpackSuccessful = "UnpackSuccessful"
+	ReasonUnpackFailed     = "UnpackFailed"
+
+	PhasePending   = "Pending"
+	PhaseUnpacking = "Unpacking"
+	PhaseFailing   = "Failing"
+	PhaseUnpacked  = "Unpacked"
+)
 
 // BundleSpec defines the desired state of Bundle
 type BundleSpec struct {
-	// Class specifies the name of the provisioner that should manage the Bundle.
-	// +optional
-	Class string `json:"class,omitempty"`
+	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
+	ProvisionerClassName string `json:"provisionerClassName"`
```

Comment:
Note: Joe's implementation has Bundle's specifying the unique provisioner ID (e.g. `kuberpak.io/registry+v1`) vs. referencing the metadata.Name of an existing ProvisionerClass resource, which has the unique provisioner ID referenced in it's spec. I think this implementation is fine for the time being as we continue to iterate and prototype, and we can re-introduce (or re-evaluate) the value of this ProvisionerClass API and make that breaking change before GA.